specs/src/specs/resource_pricing.md

```diff
@@ -0,0 +1,248 @@
+# Resource Pricing
+
+For all standard cosmos-sdk transactions (staking, IBC, etc), Celestia utilizes
+the [default cosmos-sdk mechanisms](https://docs.cosmos.network/v0.46/basics/gas-fees.html) for pricing resources. This involves
+incrementing a gas counter during transaction execution each time the state is
+read from/written to, or when specific costly operations occur such as signature
+verification or inclusion of data.
+
+```go
+// GasMeter interface to track gas consumption
+type GasMeter interface {
+	GasConsumed() Gas
+	GasConsumedToLimit() Gas
+	GasRemaining() Gas
+	Limit() Gas
+	ConsumeGas(amount Gas, descriptor string)
+	RefundGas(amount Gas, descriptor string)
+	IsPastLimit() bool
+	IsOutOfGas() bool
+	String() string
+}
+```
+
+We can see how this gas meter is used in practice by looking at the store.
+Notice where gas is consumed each time we write or read, specifically a flat
+cost for initiating the action followed by a prorated cost for the amount of
+data read or written.
+
+```go
+// Implements KVStore.
+func (gs *Store) Get(key []byte) (value []byte) {
+	gs.gasMeter.ConsumeGas(gs.gasConfig.ReadCostFlat, types.GasReadCostFlatDesc)
+	value = gs.parent.Get(key)
+
+	// TODO overflow-safe math?
+	gs.gasMeter.ConsumeGas(gs.gasConfig.ReadCostPerByte*types.Gas(len(key)), types.GasReadPerByteDesc)
+	gs.gasMeter.ConsumeGas(gs.gasConfig.ReadCostPerByte*types.Gas(len(value)), types.GasReadPerByteDesc)
+
+	return value
+}
+
+// Implements KVStore.
+func (gs *Store) Set(key []byte, value []byte) {
+	types.AssertValidKey(key)
+	types.AssertValidValue(value)
+	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostFlat, types.GasWriteCostFlatDesc)
+	// TODO overflow-safe math?
+	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostPerByte*types.Gas(len(key)), types.GasWritePerByteDesc)
+	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostPerByte*types.Gas(len(value)), types.GasWritePerByteDesc)
+	gs.parent.Set(key, value)
+}
+```
+
+The configuration for the gas meter used by Celestia is as follows.
```

Comment:
[Non-blocking]
Is there a rationale for the values used?

Reply:
yeah, there's no way to change them at the moment :laughing:

Reply:
What about the values?  Did we decide on them somewhere?

Reply:
sorry, what I meant is that we didn't discuss them or decide on them anywhere since they've never been able to be changed. These values are currently used for every cosmos-sdk chain. 

That's not to say that we shouldn't! We've occasionally mentioned that we could change these to optimize for specific things, such as aggregating data being posted or allowing many PFBs

Reply:
Aah, thanks a lot